src/basic/process-util.c

```diff
@@ -675,7 +675,7 @@ int getenv_for_pid(pid_t pid, const char *field, char **_value) {
                 }
                 line[i] = 0;
 
-                if (memcmp(line, field, l) == 0 && line[l] == '=') {
+                if (strneq(line, field, l) && line[l] == '=') {
```

Comment:
I don't have the full context here, but if the intention is to cover cases where `strlen(line) < l`, then `line[l` would still possibly be an out-of-bounds access?